src/linux_mcp_server/tools/storage.py

```diff
@@ -83,287 +98,155 @@ async def list_block_devices(
-            ["du", "-b", "--max-depth=1", path_str],
-            host=host,
-            username=username,
-        )
-
-        # Parse output - du may return non-zero on permission errors but still give valid data
-        lines = stdout.strip().split("\n")
-        dir_sizes = []
-
-        for line in lines:
-            if not line:
-                continue
-            parts = line.split("\t", 1)
-            if len(parts) == 2:
-                try:
-                    size = int(parts[0])
-                    dir_path_str = parts[1]
-                    # Skip the parent directory itself
-                    dir_name = Path(dir_path_str).name
-                    if dir_path_str != path_str:
-                        dir_sizes.append((dir_name, size))
-                except (ValueError, IndexError):
-                    continue
-
-        if not dir_sizes:
-            # Only error if we got no output AND a bad return code
-            if returncode != 0:
-                return "Error: du command failed and returned no directory data"
-            return f"No subdirectories found in: {path}"
-
-        # Sort by size (descending) and take top N
-        dir_sizes.sort(key=lambda x: x[1], reverse=True)
-        top_dirs = dir_sizes[:validated_top_n]
-
-        # Format output
-        result = []
-        result.append(f"=== Top {len(top_dirs)} Largest Directories ===")
-        result.append(f"Path: {path_str}")
-        result.append(f"\nTotal subdirectories found: {len(dir_sizes)}\n")
-
-        for i, (dir_name, size) in enumerate(top_dirs, 1):
-            result.append(f"{i}. {dir_name}")
-            result.append(f"   Size: {format_bytes(size)}")
-
-        return "\n".join(result)
-
-    except Exception as e:
-        return f"Error analyzing directories: {str(e)}"
-
-
-@mcp.tool()
+@mcp.tool(
+    title="List directories",
+    description="List directories under a specified path with flexible sorting options.",
+    annotations=ToolAnnotations(readOnlyHint=True),
+)
 @log_tool_call
-async def list_directories_by_name(
-    path: str,
-    reverse: bool = False,
-    host: str | None = None,
-    username: str | None = None,
-) -> str:
-    """
-    List directories under a specified path sorted by name.
-
-    This function uses efficient Linux primitives (find and sort) to list directories.
-
-    Args:
-        path: The directory path to analyze
-        reverse: If True, sort in reverse alphabetical order (Z-A)
-        host: Optional remote host to connect to
-        username: Optional SSH username (required if host is provided)
-
-    Returns:
-        Formatted string with directory names, or error message if validation fails
-    """
-    import os
-
+async def list_directories(  # noqa: C901
+    path: t.Annotated[str, Field(description="The directory path to analyze")],
+    order_by: t.Annotated[
+        str, Field(description="Sort order - 'size', 'name', or 'modified' (default: 'name')")
```

Comment:
Using `OrderBy` in the type hint means that validation isn't required inside the function.
```suggestion
        OrderBy, Field(description="Sort order - 'size', 'name', or 'modified' (default: 'name')")
```